wgsl/index.bs

```diff
@@ -520,6 +520,76 @@ type.
 
 ## Texture Types TODO ## {#texture-types}
 
+A <dfn noexport>texel</dfn> is a scalar or vector used as the smallest independently addressible element of a [=texture=].
+The word *texel* is short for *texture element*.
+
+A <dfn noexport>texture</dfn> is a grid of texels supporting special operations useful for rendering.
+A texture's representation is typically optimized for those operations, with many details hidden from the programmer
+including data layouts, data types, and operations that cannot be expressed directly in the shader language.
+Each texel has a unique grid coordinate within the texture.
+
+An <dfn noexport>arrayed texture</dfn> is a texture which is organized as a homegeneous array of grids of texels.
+In an arrayed texture, a texel is associated with a unique combination of array index and grid coordinate.
+
+A texture has the following features:
+
+: <dfn noexport>texel format</dfn>
+:: the data in each texel. See [[#texel-formats]]
+:  dimensionality
+:: the number of dimensions in the grid coordinates, and how the coordinates are interpreted.
+    The number of dimensions is 1, 2, or 3.
+    In some cases the third coordinate is decomposed so as to specify a cube face and a layer index.
+: size
+:: the extent of grid coordinates along each dimension
+: arrayed
+:: whether the texture is arrayed
+: <dfn noexport>array size</dfn>
+:: the number of homogeneous grids, if the texture is arrayed
+
+A shader does not have direct access to the texel storage within a texture variable.
+Instead, texture builtin functions are used to access the data.
+
+[SHORTNAME] defines texture types which control which operations are available.
+determine the set of usable builtin functions.
+
+TODO(dneto): Describe how texture types determine whether texels are readable, writeable,
+how they are addressed, and how samplers affect both addressing and filtering.
+Preview that details are provided in later sections.
+
+### Texel formats ### {#texel-formats}
+
+A [=texel format=] specifies the data contained in a [=texel=].
+
+A texel format is characterized by:
+
+: <dfn noexport>channels</dfn>
+:: Each channel contains a scalar.
+    There are up to four channels, named `r`, `g`, `b`, or `a`.
+: <dfn noexport>channel types</dfn>
+:: The number of bits in the channel, and how those bits are interpreted.
+    In some cases different channels in a texel will have different channel types.
+
+<table class='data'>
+  <caption>Channel types</caption>
+  <thead>
+    <tr><td>Channel type<td>Number of stored bits<td>Interpetation of stored bits<td>Shader type<td>Shader value
```

Comment:
I'd suggest changing this to 1 line per `td` to make it easier to read/work with

Reply:
I did that for the headings.
But I left the rows as-is.  I find it easier to both edit and scan for differences when they're horizontally organized